### ISSUE TITLE
Error: buffer overflow detected; python terminated; Aborted (core dumped)

### ISSUE THREAD

**Issue by NagabhushanSN95** [YunzhuLi/DPI-Net#4]
Hi, I installed DPI-Net and dependencies as outlined in README.md file. As I have cuda 10.2, I compiled PyFlex in docker (as suggested). But when I try to run a sample, I get the following error. Any idea/suggestions how I can resolve this?
```
$ python test_FluidFall.py
Unable to initialize SDLCould not initialize GL extensions
*** buffer overflow detected ***: python terminated
Aborted (core dumped)
```

```
$ bash scripts/eval_FluidFall.sh
Namespace(attr_dim=1, dataf='data/data_FluidFall', debug=0, dt=0.016666666666666666, env='FluidFall', epoch=12, eval=1, evalf='dump_FluidFall/eval_FluidFall', iter=180000, n_his=0, n_instance=1, n_instances=0, n_stages=1, neighbor_radius=0.08, nf_effect=200, nf_particle=200, nf_relation=300, outf='dump_FluidFall/files_FluidFall', position_dim=3, pstep=2, relation_dim=1, shape_state_dim=14, state_dim=6, time_step=121, time_step_clip=5, verbose_data=0, verbose_model=0)
Loading stored stat from data/data_FluidFall
Loading network from dump_FluidFall/files_FluidFall/net_epoch_12_iter_180000.pth
Unable to initialize SDLCould not initialize GL extensions
*** buffer overflow detected ***: python terminated
scripts/eval_FluidFall.sh: line 1: 18298 Aborted                 (core dumped) python eval.py --env FluidFall --epoch 12 --iter 180000
```

**Comment by YunzhuLi** [comment]
Is your setup able to launch OpenGL applications? When running the code, PyFleX will open an OpenGL window to show the simulation results. It will raise errors if it fails to do so.

The recommended way of playing with PyFleX is to run it on a local workstation with CUDA support (which is how I work with it). Other workarounds may work but are not guaranteed.

**Comment by NagabhushanSN95** [comment]
Yes, I tried to run a simple OpelGL program (to display some shapes) and it worked. I tried on multiple machines (they have 22GB NVIDIA Graphics Card with cuda installed) and got the same error on all machines. Do you think it might be due to mismatch in package versions? Can you please tell me the versions of the packages and softwares (like python, cudatoolkit, pytorch, torchvision, OpenGL, cuda, nvidia-driver)? I'll try to use the same versions and see if that fixes the issue.

**Comment by YunzhuLi** [comment]
I've tested the repo again, and it works as expected on my local machine: Python 3.7.4, PyTorch 1.6.0, CUDA 10.2, Nvidia driver 440.33.01, OpenGL 4.6.0. Still, this repo should work with some older or newer versions.

**Comment by YunzhuLi** [comment]
Which script produced the error? There is a chance when the code is creating the path to object files like the following, the length of the string goes beyond 100. You can try modifying the length of the char buffer to make it longer or move the PyFleX repo to a place that won't go over the length limit.
[https://github.com/YunzhuLi/PyFleX/blob/1391799c77b0bd8f5ab5a2a01f1f2ee64fb5a929/bindings/scenes/yz_boxbath.h#L33](https://github.com/YunzhuLi/PyFleX/blob/1391799c77b0bd8f5ab5a2a01f1f2ee64fb5a929/bindings/scenes/yz_boxbath.h#L33)

**Comment by NagabhushanSN95** [comment]
Ohh! This may be the reason. I'll check by making this change.
Both of the following scripts gave the error

```
cd ${PYFLEXROOT}/bindings/examples
python test_FluidFall.py
bash scripts/eval_FluidFall.sh
```

**Comment by NagabhushanSN95** [comment]
Hi, I tried the same thing by closing PyFlex in the home directory. buffer overflow is gone now. But I'm getting a new error. 

```
Unable to initialize SDLCould not initialize GL extensions
Segmentation fault (core dumped)
```

I installed OpenGL as `sudo apt-get install cmake libx11-dev xorg-dev libglu1-mesa-dev freeglut3-dev libglew1.5 libglew1.5-dev libglu1-mesa libglu1-mesa-dev libgl1-mesa-glx libgl1-mesa-dev libglfw3-dev libglfw3` following [this blog](https://medium.com/@theorose49/install-opengl-at-ubuntu-18-04-lts-31f368d0b14e). I was even able to run the sample code [here](https://www.opengl.org/archives/resources/code/samples/glut_examples/examples/examples.html).

Any ideas what might be going wrong? I couldn't find any helpful suggestions in Google either

**Comment by m2408gj** [comment]
> Hi, I tried the same thing by closing PyFlex in the home directory. buffer overflow is gone now. But I'm getting a new error.
> 
> ```
> Unable to initialize SDLCould not initialize GL extensions
> Segmentation fault (core dumped)
> ```
> 
> I installed OpenGL as `sudo apt-get install cmake libx11-dev xorg-dev libglu1-mesa-dev freeglut3-dev libglew1.5 libglew1.5-dev libglu1-mesa libglu1-mesa-dev libgl1-mesa-glx libgl1-mesa-dev libglfw3-dev libglfw3` following [this blog](https://medium.com/@theorose49/install-opengl-at-ubuntu-18-04-lts-31f368d0b14e). I was even able to run the sample code [here](https://www.opengl.org/archives/resources/code/samples/glut_examples/examples/examples.html).
> 
> Any ideas what might be going wrong? I couldn't find any helpful suggestions in Google either

Hi, have you solved this issue?

**Comment by NagabhushanSN95** [comment]
I think not. Sorry, I don't remember exactly.

**Comment by fakerloveu** [comment]
I have the same problem and I have spent more than 100h on this problem
